### PR TITLE
feat(op-dispute-mon): Enrich Claims with Resolved Status

### PR DESCRIPTION
**Pull request by refcell**
**Description**

Wrap EnrichedGameData Claims with their Resolved status and enrich this field using a new ClaimEnricher.

**Metadata**

Needed for https://github.com/ethereum-optimism/optimism/pull/9928

### CODE REVIEW

**Comment by refcell** [comment]
* **#9937** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/9937?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @refcell and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: op-dispute-mon/mon/resolution/delay.go [hunk 1]
```diff
@@ -22,25 +21,24 @@ func NewDelayCalculator(metrics DelayMetrics, clock clock.Clock) *DelayCalculato
...
 		maxDelay = max(d.getOverflowTime(game.Duration, &claim), maxDelay)
 	}
 	return maxDelay
 }
 
-func (d *DelayCalculator) getOverflowTime(maxGameDuration uint64, claim *types.Claim) uint64 {
-	// If the bond amount is the max uint128 value, the claim is resolved.
-	if monTypes.ResolvedBondAmount.Cmp(claim.ClaimData.Bond) == 0 {
+func (d *DelayCalculator) getOverflowTime(maxGameDuration uint64, claim *types.EnrichedClaim) uint64 {
+	if claim.Resolved || claim.Bond.Cmp(types.ResolvedBondAmount) == 0 {
```
Comment (Inphi) [inline op-dispute-mon/mon/resolution/delay.go]: This is fine but just a general note. What's the usecase for enriched claims enriched with optional data? Is it possible for Resolved to be true but its bond not set to the resolved amount?

Having a complete and total enriched claim ctor would simplify things. By guaranteeing the claim has all enrichments then we wouldn't have to check for the ResolvedBondAmount.

**Review by Inphi — APPROVED** [review]

**Comment by Inphi** [comment]
LGTM. Just have one small comment

File: op-dispute-mon/mon/resolution/delay.go (hunk 1 above)
Comment (refcell) [reply op-dispute-mon/mon/resolution/delay.go]: That's true, we should be able to rely on the `Resolved` field here being enriched apriori.

**Review by ajsutton — APPROVED** [review]
That was bigger than I thought it would be but I think well worth it. Thanks for ploughing through it.

**Pull request merged**